### PR TITLE
fix: require singleton GET responses to conform to JSON API

### PR DESCRIPTION
**Pull request by cmars**
Singleton resource GET responses should be required to conform to JSON API resource object requirements.

Added test coverage for this missed rule condition with a bit of refactoring to reuse test data.

### CODE REVIEW

File: end-end-tests/api-standards/resources/thing/2021-11-10/001-singleton.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ paths:
       parameters:
         - { $ref: "#/components/x-rest-common/parameters/Version" }
       responses:
-        "201":
+        "200":
```
Comment (carissadean) [inline end-end-tests/api-standards/resources/thing/2021-11-10/001-singleton.yaml:24]: Non-blocking (especially as it's not part of this PR) but the description and operationId above and below should be updated to match that this is a get request

**Review by carissadean — APPROVED** [review]

File: end-end-tests/api-standards/test-bulk.bash [hunk 2]
```diff
@@ -14,15 +14,17 @@ trap "rm -rf $tempdir" EXIT
 set +e
 
 function assert_ok {
-    if [ "$?" != "0" ]; then
-        echo "unexpected rc=$?"
```
Comment (cmars) [inline end-end-tests/api-standards/test-bulk.bash:18]: This `$?` was wrong, it was the status of the test above! That didn't affect the decision logic, but the log output was very wrong.
